pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/FieldIndexConfigsUtil.java

```diff
@@ -103,4 +106,22 @@ public <C extends IndexConfig> ColumnConfigDeserializer<C> get(IndexType<C, ?, ?
       return indexType::getConfig;
     }
   }
+
+  public static TableConfig createTableConfigFromOldFormat(TableConfig tableConfig, Schema schema) {
```

Comment:
I think the idea of having a method that upgrades a TableConfig to the latest _syntax_ is great. But I wouldn't constraint it to index related config. I would recommend to move this to one of the two TableConfigUtils.

Reply:
Makes sense, will do the changes.